traits/has_traits.py

```diff
@@ -2034,7 +2037,7 @@ def configure_traits(
                 with open(filename, "rb") as fd:
                     self.copy_traits(pickle.Unpickler(fd).load())
 
-        if edit:
+        if edit is None:
```

Comment:
I think we can refactor this a bit to remove duplication: the goal would be to not have the same logic expressed in different branches. (Read up on the "DRY Principle" for some of the rationale for avoiding duplication.)

If you do something like:

```
if edit is None:
    edit = True
else:
    warnings.warn(...)
```

here, then you should be able to leave the rest of the code unchanged.

Reply:
But as we're showing warning for both `edit=True` and `edit=False`, then refactoring like 
 ```python
 if edit is None:
     edit = True
 else:
     warnings.warn(...)
 ```
would give warning even on None, but we don't want that. 

I've added spaces as mentioned and refactored code. See if it's good?

> Sorry for being late. I was out of station for some days so I was not able to work on this.

Reply:
> would give warning even on None

I don't see how. If `edit` is `None`, then the first branch of the `if` statement is taken, and no warning is given. If `edit` is not `None`, then the `else` branch is taken, and a warning is given. I think that exactly matches what we want, doesn't it?